torch/functional.py

```diff
@@ -87,8 +86,28 @@ def split(tensor, split_size_or_sections, dim=0):
     # call here.
     return tensor.split(split_size_or_sections, dim)
 
+# equivalent to itertools.product(indices)
+def indices_product(indices):
+    # type: (List[int]) -> (List[List[int]])
+    empty_list : List[int] = []
```

Comment:
We still have to support python 2 so I think you have to use `torch.jit.annotate`